crates/edr_instrument/src/coverage.rs

```diff
@@ -0,0 +1,212 @@
+use edr_eth::B256;
+pub use semver::Version;
+use serde::Serialize;
+use sha3::{Digest, Keccak256};
+use slang_solidity::{
+    cst::{Node, NonterminalKind},
+    parser::{ParseError, Parser as SolidityParser, ParserInitializationError},
+};
+
+#[derive(Serialize)]
+pub struct InstrumentationMetadata {
+    pub tag: B256,
+    pub kind: &'static str,
+    pub start_utf16: usize,
+    pub end_utf16: usize,
+}
+
+pub struct InstrumentationResult {
+    pub source: String,
+    pub metadata: Vec<InstrumentationMetadata>,
+}
+
+/// Computes a deterministic hash for a statement in a Solidity file.
+///
+/// For the time being, we're assuming that compilation configuration has no
+/// impact on the statement hash. This is a simplification, but is considered a
+/// reasonable trade-off for the current use case.
+fn compute_deterministic_hash_for_statement(
+    source_id: &str,
+    content_hash: &B256,
+    statement_counter: u64,
+) -> B256 {
+    let mut hasher = Keccak256::new();
```

Comment:
I'm curious about the choice of the hash function. Is the purpose just to generate unique ids deterministically? If yes, we could consider using Blake3 instead of Keccak, as it's an order of magnitude faster: https://github.com/BLAKE3-team/BLAKE3?tab=readme-ov-file#blake3

Reply:
I started a thread to discuss this internally. If we agree to move forward with Blake3 (an idea I'm all for based on your description) I'll make that change in a follow-up PR such that the Hardhat team can start working with the API earlier.